src/NJsonSchema/Visitors/AsyncJsonReferenceVisitorBase.cs

```diff
@@ -136,10 +136,10 @@ protected virtual async Task VisitAsync(object obj, string path, string typeName
                     await VisitAsync(schema.AnyOf.ElementAt(i), path + "/anyOf[" + i + "]", null, checkedObjects, o => ReplaceOrDelete(schema.AnyOf, index, (JsonSchema)o), cancellationToken).ConfigureAwait(false);
                 }
 
-                for (var i = 0; i < schema.OneOf.Count; i++)
+                for (var i = 0; i < schema._oneOf.Count; i++)
```

Comment:
Why do you change that? 
Is there a perf difference? It should be inlined in the old form, no?

Reply:
(Here's IL comparison)[https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgBgAJDSUBuK2h0gOgGEJdcYYDNggA7AM6cA8sABWgjAFkIAExi5WlNgGYGxOtzoBvKnTN1T5gMpgAFjAC2AQzri7jlwF46ogK79NcwtKIMJdQhQ6AElRDBgoADMnMBgACgBKYKCTEKDzBOg6VIA3Jyg6bDpvGmYKugAeV3dnaVEYKQSeCF9Y2uwAan70yzzjEbyAX3GprVyzMIZI3lEwWDiMrPMc0bMC8pKyuuq+hqb7FoB9MXbO3h6MPsHhubztnaDxoJmv8d+XhYiBjEqxgcQud1i3GS9hUG3Gb1GpXKkHuVTOHk4VzaHS690COz2RSRRzoNTqjRRvQqT0+W1po3pdG+5hmVFZ1HC+hs5ycVARFVi8VETlwdBk4nipWAAl4/AUIlE9UYNAAfHQsTd8WZxgsorKBEIFUrSKqxdiEvDGQBzUFVNUajpapkfSgTIA] so as seemed to be in hot path (with Fluid as actually can see something in profiler outside of DotLiquid). The samples show first how two virtual calls in loop, then one with new code and third alternative where no virtual calls in loop. 

This is still work in progress and hard to see difference against current master as all time is spent in DotLiquid.

Reply:
And by all means this is not the biggest thing to consider with. I just tend to see it happen a lot of libraries that fields can be declared as `IDictionary<string, object> _theField = new Dictionary<string, object>();` and after that everything is virtual dispatch slowing things down. [I worked extensively with Jint](https://github.com/sebastienros/jint/issues/451#issuecomment-412627408) optimizing all kinds of access patterns.

Reply:
> This is still work in progress and hard to see difference against current master as all time is spent in DotLiquid.

So we'd first move to Fluid without this PR and then do optimizations on top of Fluid?

Reply:
Ideally if possible, Fluid makes it *really fast* compared to baseline and that would be come the new baseline which would be easier to do targeted optimizations against.